test/src/test/java/org/corfudb/runtime/clients/NettyCommTest.java

```diff
@@ -569,6 +573,141 @@ public void testTlsCipherRsaAndEcdsaKeyStoreEcdsaEcdsa() throws Exception {
                 true, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384);
     }
 
+    @Test
+    public void testBasicReloadSslCerts() throws Exception {
+        int port = findRandomOpenPort();
+
+        Path certDir = Paths.get(PARAMETERS.TEST_TEMP_DIR);
+
+        // Start corfu server
+        CertificateManager serverCertManager = CertificateManager.buildSHA384withEcDsa(certDir);
+        NettyServerData serverData = new NettyServerData(buildServerContext(port, serverCertManager.certManagementConfig));
+        serverData.bootstrapServer();
+
+        // Happy path
+        CertificateManager clientCertManager = CertificateManager.buildSHA384withEcDsa(certDir);
+        clientCertManager.trustStoreManager.addCertificate(serverCertManager);
+        clientCertManager.trustStoreManager.save();
+        serverCertManager.trustStoreManager.addCertificate(clientCertManager);
+        serverCertManager.trustStoreManager.save();
+        NettyClientRouter clientRouter = new NettyClientRouter(
+                NodeLocator.builder().host("localhost").port(port).build(),
+                buildRuntimeParams(clientCertManager.certManagementConfig)
+        );
+        clientRouter.getConnectionFuture().join();
+        assertThat(getBaseClient(clientRouter).pingSync()).isTrue();
+
+        // Reload ssl
+        clientRouter.reconnect();
```

Comment:
Can we wait for a timeout of `fileWatcherPollPeriod` for the `clientRouter` to be reconnected by the Filewatcher service's `onChange.run()` instead of calling `reconnect()` explicitly?

Reply:
Good point. I will add a test for the FileWatcher logic. I will keep them as two separate tests. Thank you.